### PR TITLE
Create a troubleshooting guide for high scale deployments

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

**Comment by jasonblais** [comment]
@cwarnermm or @justinegeffen would one of you be open to review some time today?

File: source/deploy/high-scale-troubleshoot.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+.. contents::
+  :backlinks: top
+  :local:
+
+My system keeps hanging when I search for a message in Mattermost
+---------------------------------------------------------------------
+
+First, check how many messages have been posted on your system, including deleted posts and posts made using automations.
+
+Go to the **System Console > Reporting > Site Statistics** and review the **Total Posts** figure reported. If it is above 3,000,000 posts, we recommend deploying Elasticsearch alongside your Mattermost server for improved search performance. Follow our guides to `deploy an elasticsearch server <https://docs.mattermost.com/scale/elasticsearch.html>`_.
```
Comment (cwarnermm) [inline source/deploy/high-scale-troubleshoot.rst:19]:
```suggestion
Go to the **System Console > Reporting > Site Statistics** and review the **Total Posts** figure reported. If it is above 3,000,000 posts, we recommend deploying Elasticsearch alongside your Mattermost server for improved search performance. Follow our guides to `deploy an Elasticsearch server </scale/elasticsearch.html>`__.
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @jasonblais!

**Comment by jasonblais** [comment]
Thank you Carrie!

**Pull request merged**